### PR TITLE
[Sub-ports] added basic sub-ports tests

### PR DESCRIPTION
**Pull request by OleksandrKozodoi**
Signed-off-by: Oleksandr Kozodoi <oleksandrx.kozodoi@intel.com>

### Description of PR

Summary:
Added new basic sub-ports tests.
Test suite will be enhanced with new cases asap.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
Coverage of sub-ports feature by test cases to improve the quality of SONiC.
#### How did you do it?
Added new test cases to PyTest.
Implementation is based on the [Sub-ports design spec](https://github.com/Azure/SONiC/blob/master/doc/subport/sonic-sub-port-intf-hld.md)
#### How did you verify/test it?
`py.test --testbed=testbed-t0 --inventory=../ansible/lab --testbed_file=../ansible/testbed.csv --host-pattern=testbed-t0 -- 
module-path=../ansible/library sub_port_interfaces`
#### Any platform specific information?
SONiC Software Version: SONiC.master.35-dirty-20201112.031542
Distribution: Debian 10.6
Kernel: 4.19.0-9-2-amd64
Build commit: 6c362a08
Build date: Thu Nov 12 11:49:55 UTC 2020
Platform: x86_64-accton_wedge100bf_32x-r0
HwSKU: montara
ASIC: barefoot
#### Supported testbed topology if it's a new test case?
T0 only

### CODE REVIEW

**Comment by wangxin** [comment]
retest vsimage please

File: tests/sub_port_interfaces/conftest.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    """
+    sub_ports_vars = {
+        'sub_ports': define_sub_ports_configuration['sub_ports'],
+    }
+
+    duthost.host.options['variable_manager'].extra_vars.update(sub_ports_vars)
+
+    duthost.command("mkdir -p {}".format(DUT_TMP_DIR))
+    sub_ports_config_path = os.path.join(DUT_TMP_DIR, SUB_PORTS_TEMPLATE)
+    duthost.template(src=os.path.join(TEMPLATE_DIR, SUB_PORTS_TEMPLATE), dest=sub_ports_config_path)
```
[the remark this answers is not in the document]
Comment (OleksandrKozodoi) [reply tests/sub_port_interfaces/conftest.py]: Great suggestion. Updated. Thanks.

File: tests/sub_port_interfaces/Sub-ports-test-plan.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+|Ethernet4.10|172.16.0.1/30|eth1.10     |172.16.0.2/30|
+|Ethernet4.20|172.16.0.5/30|eth1.20     |172.16.0.6/30|
+|Ethernet8.10|172.16.4.1/30|eth2.10     |172.16.4.2/30|
+|Ethernet8.20|172.16.4.5/30|eth2.20     |172.16.4.6/30|
+
+After end of the test session teardown procedure turns testbed to the initial state.
+
+## Python scripts to setup and run test
+
+Sub-ports test suite is located in tests/sub_port_interfaces folder. The is one files test_sub_port_interfaces.py
```
[the remark this answers is not in the document]
Comment (OleksandrKozodoi) [reply tests/sub_port_interfaces/Sub-ports-test-plan.md]: Updated. Thanks.

File: tests/sub_port_interfaces/conftest.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+import pytest
+
+from tests.common import config_reload
+from sub_ports_helpers import DUT_TMP_DIR
+from sub_ports_helpers import TEMPLATE_DIR
+from sub_ports_helpers import SUB_PORTS_TEMPLATE
+
+
+@pytest.fixture()
```
[the remark this answers is not in the document]
Comment (OleksandrKozodoi) [reply tests/sub_port_interfaces/conftest.py]: Done. Thanks.

File: tests/sub_port_interfaces/conftest.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    sub_ports_config_path = os.path.join(DUT_TMP_DIR, SUB_PORTS_TEMPLATE)
+    duthost.template(src=os.path.join(TEMPLATE_DIR, SUB_PORTS_TEMPLATE), dest=sub_ports_config_path)
+
+    duthost.command('sonic-cfggen -j {} --write-to-db'.format(sub_ports_config_path))
+
+    yield sub_ports_vars
+    reload_dut_config(duthost)
+
+
+@pytest.fixture()
```
[the remark this answers is not in the document]
Comment (OleksandrKozodoi) [reply tests/sub_port_interfaces/conftest.py]: Done. Thanks.

**Review by wangxin — APPROVED** [review]

**Comment by OleksandrKozodoi** [comment]
@wangxin @yxieca Do we have some obstacles for this PR not to be merged?

**Comment by lguohan** [comment]
@wangxin , what is the concern for not merging it?

**Pull request merged**